File: session-20161122/src/FizzBuzz.java
public class FizzBuzz {
    private final int BUZZ = 5;
    private final int FIZZ = 3;
    private final int FIZZBUZZ = 15;

    public String computeFizzBuzz(int number) {
        if (number % FIZZBUZZ == 0) {
            return "FizzBuzz";
        }

        if (number % BUZZ == 0) {
            return "Buzz";
        }

        if (number % FIZZ == 0) {
            return "Fizz";
        }

        return String.valueOf(number);
    }
}

File: session-20170418/src/FizzBuzz.ts
import * as _ from 'lodash';

export const FIZZ = 'fizz';
export const BUZZ = 'buzz';
export const FIZZBUZZ = 'fizzbuzz';

export function getFizzBuzzRange(from: number, to: number): string[] {
    return _.range(from, to + 1).map(getFizzBuzz);
}

export function getFizzBuzz(value: number): string {
    let isFizz = value % 3 === 0 || `${value}`.includes('3');
    let isBuzz = value % 5 === 0 || `${value}`.includes('5');

    if (isFizz && isBuzz)
        return FIZZBUZZ;

    if (isBuzz)
        return BUZZ;

    if (isFizz)
        return FIZZ;

    return `${value}`;
}
